usb/esp_tinyusb/tusb_cdc_acm.c

```diff
@@ -382,26 +382,34 @@ esp_err_t tusb_cdc_acm_init(const tinyusb_config_cdcacm_t *cfg)
     acm->ringbuf_read_mux = xSemaphoreCreateMutex();
     if (acm->ringbuf_read_mux == NULL) {
         ESP_LOGE(TAG, "Creation of a ringbuf mutex failed");
-        free_obj(itf);
-        return ESP_ERR_NO_MEM;
+        ret = ESP_ERR_NO_MEM;
+        goto clean;
     }
 
     acm->rx_tfbuf = malloc(CONFIG_TINYUSB_CDC_RX_BUFSIZE);
     if (!acm->rx_tfbuf) {
         ESP_LOGE(TAG, "Creation buffer error");
-        free_obj(itf);
-        return ESP_ERR_NO_MEM;
+        ret = ESP_ERR_NO_MEM;
+        goto clean;
     }
     acm->rx_unread_buf_sz = cfg->rx_unread_buf_sz == 0 ? RX_UNREADBUF_SZ_DEFAULT : cfg->rx_unread_buf_sz;
     acm->rx_unread_buf = xRingbufferCreate(acm->rx_unread_buf_sz, RINGBUF_TYPE_BYTEBUF);
     if (acm->rx_unread_buf == NULL) {
         ESP_LOGE(TAG, "Creation buffer error");
-        free_obj(itf);
-        return ESP_ERR_NO_MEM;
+        ret = ESP_ERR_NO_MEM;
+        goto clean;
     } else {
         ESP_LOGD(TAG, "Comm Initialized buff:%d bytes", cfg->rx_unread_buf_sz);
         return ESP_OK;
     }
+clean:
+    if (acm->rx_tfbuf) {
+        free(acm->rx_tfbuf);
+        acm->rx_tfbuf = NULL;
+    }
+    free_obj(itf);
+    tinyusb_cdc_deinit(itf);
+    return ret;
```

Comment:
Should we unregister the callbacks too?

Reply:
No need.
On registering callbacks, these are assigned in CDC_ACM object esp_tusb_cdcacm_t. 
free_obj(itf); actually frees up the memory allocated by esp_tusb_cdcacm_t struct. So, unregistering of callbacks are not needed because memory is getting freed up.